lib/board.rb

```diff
@@ -0,0 +1,54 @@
+class Board
+  require 'pry'
+  attr_accessor :cells, :board
+
+  def initialize
+    @board = Array.new(9, " ")
+    @cells = @board
+  end
+
+  def reset!
+    @board = Array.new(9, " ")
+    @cells = Array.new(9, " ")
+  end
+
+  def display
+    puts " #{self.cells[0]} | #{self.cells[1]} | #{self.cells[2]} "
+    puts "-----------"
+    puts " #{self.cells[3]} | #{self.cells[4]} | #{self.cells[5]} "
+    puts "-----------"
+    puts " #{self.cells[6]} | #{self.cells[7]} | #{self.cells[8]} "
+  end
+
+  def position(num)
+    self.cells[num.to_i - 1]
+  end
+
+  def full?
+    self.cells.all? {|cell| cell == "O" || cell == "X"}
+  end
+
+  def turn_count
+    count = 0
+
+    self.cells.each do |cell|
+      if cell == "X" || cell == "O"
+        count += 1
+      end
+    end
+    count
```

Comment:
Use the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count {|cell| cell == "O" || cell == "X"}
end
```